dao/src/main/java/org/thingsboard/server/dao/util/DeviceConnectivityUtil.java

```diff
@@ -171,6 +178,40 @@ public static String getCoapPublishCommand(String protocol, String host, String
 
     public static String getDockerCoapPublishCommand(String protocol, String host, String port, DeviceCredentials deviceCredentials) {
         String coapCommand = getCoapPublishCommand(protocol, host, port, deviceCredentials);
-        return coapCommand != null ? String.format("%s%s%s", DOCKER_RUN + (LOCAL_HOSTS.contains(host) ? "--network=host ":""), COAP_IMAGE, coapCommand) : null;
+        return coapCommand != null ? String.format("%s%s%s", DOCKER_RUN + (isLocalhost(host) ? "--network=host " : ""), COAP_IMAGE, coapCommand) : null;
+    }
+
+    public static String getHost(String baseUrl, DeviceConnectivityInfo properties, String protocol) throws URISyntaxException {
+        String initialHost = properties.getHost().isEmpty() ? baseUrl : properties.getHost();
+        InetAddress inetAddress;
+        String host = null;
```

Comment:
host = initialHost; to delete if (host == null) block

Reply:
Unfortunately, in this case we may get a null pointer exception, for example if user set his host to "http://::1", it is a bad address, but user should not care about this. After getting null we will try to operate with original incoming host and parse it using InetAddress, it may parse it correctly. Situation when user can forget to add `[...]` for ipv6 in baseUrl/host address will cause host = null in URI as well.